### ISSUE TITLE
Redesign and improve the VM Settings: Applications tab

### ISSUE THREAD

**Issue by marmarta** [QubesOS/qubes-issues#6366]
**Qubes OS version (if applicable)**
R.4.1

**Brief summary**
Make it more flexible (including app icons, open-in-dispvm setting and possibly more.

In particular: include way to use https://github.com/QubesOS/qubes-core-agent-linux/pull/290

**Comment by marmarta** [comment]
Current proposal: 
![Screenshot_2021-01-26_16-45-16](https://user-images.githubusercontent.com/7242298/105868513-8a311c80-5ff6-11eb-917f-e1eddacb66f0.png)

**Comment by marmarta** [comment]
@DemiMarie , @andrewdavidwong , @marmarek , any critique/opinions? The former tooltip contents (incl. .desktop file name) are the same.
I plan to add icons in next iteration.

**Comment by DemiMarie** [comment]
@marmarta Looks good!  I wonder if “Open Files in a Disposable VM” is a bit too concise, though.  What about, “Should this application open files in a Disposable VM?”   If that is too big for the actual display, could it be in some sort of discoverable help?

**Comment by marmarta** [comment]
> @marmarta Looks good! I wonder if “Open Files in a Disposable VM” is a bit too concise, though. What about, “Should this application open files in a Disposable VM?” If that is too big for the actual display, could it be in some sort of discoverable help?

I can add a tooltip to the header. I could also add a paragraph describing what's going on in this tab between the refresh button and the table. Hmm, that might be a good idea...

**Comment by andrewdavidwong** [comment]
> I wonder if “Open Files in a Disposable VM” is a bit too concise, though. What about, “Should this application open files in a Disposable VM?”

I'm still not sure what exactly this means. For example, let's say I check this box next to Gedit. What actually happens? Does it mean that, whenever I open a file with Gedit, it gets opened in a new DisposableVM (in which Gedit starts)? What if I already have Gedit open in the AppVM, then I use "File > Open..." inside of Gedit? That instance of Gedit is already running in the AppVM, not in a DisposableVM.

**Comment by andrewdavidwong** [comment]
> any critique/opinions? The former tooltip contents (incl. .desktop file name) are the same.
> I plan to add icons in next iteration.

If not already possible, it might be nice to be able to sort the columns. For example, a user might want to sort the applications alphabetically or sort all currently checked boxes to the top.

@ninavizz might be able to offer better UX feedback.

**Comment by marmarek** [comment]
> For example, a user might want to sort the applications alphabetically or sort all currently checked boxes to the top.

Sorting is already possible, I think the current draft (newer than the screenshot above) is sorted by "checked" by default.

**Comment by marmarta** [comment]
Current draft: 
![settings-apps2](https://user-images.githubusercontent.com/7242298/106142887-54b13e00-6172-11eb-9c7e-014cb83ac636.png)

**Comment by marmarta** [comment]
(Icons are ATM only for previously-selected apps but this will improve)

**Comment by marmarta** [comment]
> > I wonder if “Open Files in a Disposable VM” is a bit too concise, though. What about, “Should this application open files in a Disposable VM?”
> 
> I'm still not sure what exactly this means. For example, let's say I check this box next to Gedit. What actually happens? Does it mean that, whenever I open a file with Gedit, it gets opened in a new DisposableVM (in which Gedit starts)? What if I already have Gedit open in the AppVM, then I use "File > Open..." inside of Gedit? That instance of Gedit is already running in the AppVM, not in a DisposableVM.

I have an updated description in the screenshot above, is this a bit clearer for you?

**Comment by deeplow** [comment]
Question: what happens if the Default DisposableVM Template is not based on the same TemplateVM as the AppVM (i.e. may not have the same programs installed)?

**Comment by ninavizz** [comment]
- I would reverse the ordering of the "Description" and "Open in DispVM" columns. Right now it appears like there's 2 grouped-columns of selectable things... then it's confusing that these two groups of selectable things have different categories?
- Icon column, I'm not terribly keen on; adds clutter w/o clear value. Also confusing that icons are all colored blue, in a yellow VM.
- Progressive-disclosure pattern for the "Experimental" text, I recommend. Can mock something up for ya.
- A lot of characters and words. I can also think about 'cleaner' language. 

Open Question: I'm trying to install Signal in one of my Qubes. I followed the instructions on the Signal website, downloaded and did all the things from the Firefox in that Qube, and Signal is still not showing-up in the list. I want a button or informational panel that can open from here, to walk me through what it is that I'm missing that I need to do. Yes, I've rebooted many times.

**Comment by DemiMarie** [comment]
> * Not clear why there's an "Ok" and "Apply" button. I'm assuming that might be a Linux/Windows convention, but to keep it simple I'd opt to just do a clean Cancel/Ok pairing of buttons.

It’s the difference between “save” (Apply) and “save and exit” (Ok), which might be better names.

**Comment by ninavizz** [comment]
> It’s the difference between “save” (Apply) and “save and exit” (Ok), which might be better names.

Yes, I understand that; I just question the value of that many buttons, vs the clearer 2-button binary. The 3-button pattern is increasingly falling out of favor as more modern and simpler paradigms have become more dominant. 

That said, my Qubes machine is shut down and I'd also not want to introduce a new pattern wildly different from all others, if the 3-button pattern is common elsewhere.

**Comment by deeplow** [comment]
> * 
> 
> * Icon column, I'm not terribly keen on; adds clutter w/o clear value. Also confusing that icons are all colored blue, in a yellow VM.

Maybe the icon can just on the "Application" collumn, just before the name. Not sure how hard Qt can make this.

**Comment by ninavizz** [comment]
> Maybe the icon can just on the "Application" collumn, just before the name. Not sure how hard Qt can make this.

I question the value of showing icons at all on this panel, but if they are shown they should be structured in their own column; just not with a sortable/visible header, and the checkboxes should be to the left of the icons. Can do a quicky mockup tonite.

**Comment by ninavizz** [comment]
@marmarta FWIW, I'd MUCH rather work on anything Qubes atm, than the wretched project from 2 months ago that I've been fighting to close out for the past 2 weeks. Really, today has got to be the day I wrap it, so I can come back to Qubesland...

**Comment by marmarta** [comment]
> > Maybe the icon can just on the "Application" collumn, just before the name. Not sure how hard Qt can make this.
> 
> I question the value of showing icons at all on this panel, but if they are shown they should be structured in their own column; just not with a sortable/visible header, and the checkboxes should be to the left of the icons. Can do a quicky mockup tonite.

Yeah, the header should be empty (this is easy); I can move checkboxes, also easy. I don't think a mockup is necessary (buuut if you will have some time there's the whole menu stuff... ;) )

**Comment by marmarta** [comment]
> > It’s the difference between “save” (Apply) and “save and exit” (Ok), which might be better names.
> 
> Yes, I understand that; I just question the value of that many buttons, vs the clearer 2-button binary. The 3-button pattern is increasingly falling out of favor as more modern and simpler paradigms have become more dominant.
> 
> That said, my Qubes machine is shut down and I'd also not want to introduce a new pattern wildly different from all others, if the 3-button pattern is common elsewhere.

Yeah, we do use that pattern, and it's fairly useful because some changes may have lasting effects, applying other changes can influence other things etc. - also, it's nice to say "ok, current changes are fine, let me save the work before doing more changes". It's also a common pattern among Linux (and Windows, for the matter) tools.

**Comment by ninavizz** [comment]
> Yeah, we do use that pattern, and it's fairly useful because some changes may have lasting effects, applying other changes can influence other things etc. - also, it's nice to say "ok, current changes are fine, let me save the work before doing more changes". It's also a common pattern among Linux (and Windows, for the matter) tools.

This is a frequent topic of discussion at design conferences, with UX researchers and practicioners shuddering everywhere and Microsoft people quietly strolling away while whistling and staring at the floor... lol. If it's a common pattern in Qubes, then users will expect it and such I say stay! I just won't tell my friends who've wrote entire thinkpieces, attempting to bury the paradigm, hehe.


> Yeah, the header should be empty (this is easy); I can move checkboxes, also easy. I don't think a mockup is necessary (buuut if you will have some time there's the whole menu stuff... ;) )

The menu stuff is absolutely thee priority, especially per emails last week. You shall receive PDFs within a few days, and my brain very much looks forward to returning to its interaction design happyplace. :)

**Comment by andrewdavidwong** [comment]
Some notes (mostly on language):

 - Let's please use our own terminology more consistently (#4723). It should be `DisposableVM`, not `Disposable VM` or `DispVM`.
 - "in a Qubes menu or desktop shortcut" is confusing, since there is nothing called "the Qubes menu." I think the former is referring to the "Applications Menu" (as Xfce calls it).
 - There should be no comma in "...shortcut, and performs..." (since what comes after the "and" is not an independent clause).
 - "This feature is experimental and may not always work" is a turn-off to users seeking a stable, reliable system. Such users do not want to mess around with something that may not work. I suggest hiding this feature by default and perhaps making it opt-in for users who wish to experiment.

**Comment by ninavizz** [comment]
>     Let's please use our own terminology more consistently (#4723). It should be DisposableVM, not Disposable VM or DispVM.

As a label for a table, all of the above are too-long in character counts. `Open all files </br>in DVM? (?)` is how I'd personally recommend.  I do agree with the spirit of Andrew's ask, though. I also kinda wish we could standardize around `DispVM` as language used w/in the UI, but `Disposable VM` as language in docs. But, that's a different convo.

@marmarta I would like to propose some polish-suggestions for how to execute the table, so that everything is more legible and refined—but none of those suggestions would impact core interaction, I suspect. And, yes, AppMenu stuff is a priority. 

>     "in a Qubes menu or desktop shortcut" is confusing, since there is nothing called "the Qubes menu." I think the former is referring to the "Applications Menu" (as Xfce calls it).
>     There should be no comma in "...shortcut, and performs..." (since what comes after the "and" is not an independent clause).

I also think we need to get more consistent with what we call different parts of the UI; such as the "App Menu" or the "Domains Widget." 

>     "This feature is experimental and may not always work" is a turn-off to users seeking a stable, reliable system. Such users do not want to mess around with something that may not work. I suggest hiding this feature by default and perhaps making it opt-in for users who wish to experiment.

To this point, I also add that in the Qubes universe—even things that are called "stable," often are not... for reasons. That said, I also agree with what Andrew says, in spirit, while also feeling there should be a way to communicate the general idea while encouraging feedback. This imho should be done with a recognizable and linkable word/icon combo presented in documentation and on the website to denote Qubes features are are experimental in _concept_ but not execution. Like, just the word "experimental" connotes so many different things, and we need to be clear about what is meant by "experimental."

**Comment by marmarek** [comment]
The thing about experimental, is that the method how it's done may not work with some applications. If it works with a particular application, it will work reliably there. But since accidentally opening a file in a qube directly instead of a DisposableVM may have security consequences, we should clearly mark it.
In the future I think we can either get it to the state where it works with all the applications (users reports are essential to get to that state!), or have a catalog of applications compatible with this feature.

**Comment by andrewdavidwong** [comment]
> As a label for a table, all of the above are too-long in character counts. `Open all files </br>in DVM? (?)` is how I'd personally recommend.

It was previously decided that `DVM` is a bad acronym and have been migrating away from it for a long time now. I personally don't really care which name we choose, as long as we use it consistently.

> I also kinda wish we could standardize around `DispVM` as language used w/in the UI, but `Disposable VM` as language in docs.

I think that kind of inconsistency would be very confusing to users. Moreover, it was previously decided that `DispVM` is a bad name (since it can be misinterpreted as "DisplayVM"), so we've been migrating away from it for a long time now.

**I'm not arguing that any of these terms are bad. I'm merely reporting project history with respect to these terms.**

**Comment by ninavizz** [comment]
@andrewdavidwong Context and background are always super helpful, TY!! Now that we're developing a framework to solicit feedback from users in a structured fashion, these are definitely questions I'd like to explore in ongoing testing. 

@marmarek That info is also nicely helpful. Curious to dig into that some more. When I don't have other, urgent-deadline things looming... such as... oh, you know. :)

**Comment by GWeck** [comment]
It might be a good idea to apply this new design to the VM selection dialogue of the backup/retore utility, too.

**Comment by marmarta** [comment]
> It might be a good idea to apply this new design to the VM selection dialogue of the backup/retore utility, too.

It would need some adjustments (like a select all/unselect all buttons), but it might be a good idea. The "move to the right" widget we've been using is a pretty unusual design choice, I guess.

**Comment by GWeck** [comment]
If you are thinking of select all/unselect all buttons: Why not include predfined selections like in the R4.1 Qube manager: all templates / AppVMs / service VMs, or so?

**Comment by ninavizz** [comment]
OMG THE "REFRESH APPLICATIONS" BUTTON IS EVERYTHING!!

Actually... can it just automagically do that whenever the window is opened? But, yes, THANK YOU for that button! Design for menu customization for the app menu will be similar to this and to an Ubuntu menu customization window I've seen... will be nice to see all this coalesce. :)

**Comment by marmarta** [comment]
> OMG THE "REFRESH APPLICATIONS" BUTTON IS EVERYTHING!!
> 
> Actually... can it just automagically do that whenever the window is opened? But, yes, THANK YOU for that button! Design for menu customization for the app menu will be similar to this and to an Ubuntu menu customization window I've seen... will be nice to see all this coalesce. :)

Refresh apps button already exists :) And no, we don't want to do it automagically, because refresh will start up the VM's template, and that's resource-intesive - I don't think it's a good idea for VMs to suddenly start just because you looked at the settings :)

**Comment by ninavizz** [comment]
One additional question: what purpose does the "description" field serve? Is that only visible to users outside of this UI, within the _Application Search_ applet? Depending upon how it is used, it might be nice to make this editable. Just a thought. :)

**Comment by andrewdavidwong** [comment]
IMHO, it might be more intuitive to put the pin, favorite, and maybe even DisposableVM options in the app menu itself rather than here, for example, being able to right-click an entry in the app menu and having an option to pin it.

**Comment by ninavizz** [comment]
@andrewdavidwong I think both paths are important. Not all users are habitual right click users... notably, Mac mice have never had a "Right Click" button on them, so folks who either came of age digitally with Macs, or are current Mac users, never developed that habit. Which is a lot of users. Even many Windows users never develop the habit. But I still feel Right Click needs to exist, for those of us for whom it is intuitive, and power users expecting speed. 

Language was crafted very specifically to facilitate UX best practices. Active voice is important, as is leading with the action you want the person to do (vs intro words or "please;" users prefer text just getting to the point in UIs). Likewise, using similar "patterns" of language to communicate related functionality. Within those best practice values, I'm always happy to iterate! Critique isn't "qualmy," it's a must for high quality outputs. You also kinda "own" the voice and language standards around everything Qubes, so your feedback and ending-up on things we're both happy with, is important! I also have wretched grammar skills, I realize. PhD, meet BFA. :D

**Comment by andrewdavidwong** [comment]
> @andrewdavidwong I think both paths are important. Not all users are habitual right click users... notably, Mac mice have never had a "Right Click" button on them, so folks who either came of age digitally with Macs, or are current Mac users, never developed that habit. Which is a lot of users. Even many Windows users never develop the habit. But I still feel Right Click needs to exist, for those of us for whom it is intuitive, and power users expecting speed.
> 

I think it would be fine to have both. For example, Windows 10 provides multiple ways to pin an app to the Start Menu (...though perhaps some might dislike this example).

> Language was crafted very specifically to facilitate UX best practices. Active voice is important, as is leading with the action you want the person to do (vs intro words or "please;" users prefer text just getting to the point in UIs). Likewise, using similar "patterns" of language to communicate related functionality. Within those best practice values, I'm always happy to iterate! Critique isn't "qualmy," it's a must for high quality outputs. You also kinda "own" the voice and language standards around everything Qubes, so your feedback and ending-up on things we're both happy with, is important! I also have wretched grammar skills, I realize. PhD, meet BFA. :D

No worries, my "qualms" are mostly about using consistent terminology (see https://github.com/QubesOS/qubes-issues/issues/4723) instead of having "Disp-VM," "DispVM," "dispvm," "DisposableVM," "Disposable VM," etc. I didn't want to bog down the process by bringing that stuff up too early, but I also didn't want to let it go live without providing that feedback, so I just wanted to signal that I have feedback to give in that department whenever the time is right. :slightly_smiling_face:

**Comment by deeplow** [comment]
![colorize](https://user-images.githubusercontent.com/47065258/118001403-3abcf500-b336-11eb-94e3-3686289fc7d1.png)

Was the opacity selection a result of user feedback? I feel like this approach is too granular. I would suggest instead an option in the qubes global settings (or menu settings) called "colorize applications". Just with a toggle. No %. It then would apply to all qubes.

**Comment by ninavizz** [comment]
@andrewdavidwong It is a UX best practice to addend words for things like table headers and buttons. The full text "Disposable VM" is problematic as a full table header or button because of its lengthy character-count, and yet "DispVM" or "DVM" are problematic in regular writing contexts because they are unclear. In a UI (vs in a paragraph) word length matters, both for the increased cognitive friction in users making decisions in the context of other tasks, and with placement on the objects. Obvs, this is a bigger discussion than just for this pane, and I'd love to have that discussion now, so that we can clarify the standard to apply across the board. 

In the black bubbles, my suggestion is to dynamically input the named disp-vm from the user's chosen setting on another pane; not to put the word dispvm there. :)

@deeplow Yes, I believe there should also be in the Applications Menu Preferences pane as a _global_ option, but when it is enabled we then introduce the usability problems with today's feature: different colors obfuscate app icons, differently. That's just the nature of a global aesthetic treatment applied to a multitude of different backgrounds and overlay colors. 

Today's coloration of icons is a very problematic "device" from a usability perspective, and yet is a critically important security vehicle for current users who've already trained themselves to today's Qubes UI quirks. Sure, we could just not do icon overlay coloration and do little dots next to each icon, instead—but in the appmenu that becomes quite busy, so more holistic solutions were offered in the new design options. And understandably, such a significant shift in semiotic markers from one place to another, felt problematic to many users. Do you design to meet existing user needs, or to suit product growth and new user needs? This feature would help meet both needs.

I also do not know when Marta might be able to implement the global-level feature, so until then this could still help ease the burdens with today's coloration.  It is also common for accessibility and usability, when an OS does opacity "things," to give users agency in choosing their opacity.

**Comment by ninavizz** [comment]
Edited (again!) to remove "VM" nomenclature from the UI. Updated layout, too. 

![AppSettings-notips3](https://user-images.githubusercontent.com/8262612/118072031-dca40800-b35d-11eb-8a5b-d4a8e9c3ee86.png)

![AppSettings-tips3](https://user-images.githubusercontent.com/8262612/118072041-e62d7000-b35d-11eb-9f3c-c6f5eb298c03.png)

**Comment by DemiMarie** [comment]
@ninavizz Can I just say WOW???  What you are talking about just seems like magic to us tech types :smile:.  Seriously, thank you SO MUCH for your hard work!

**Comment by GWeck** [comment]
One more idea about colourizing the icons: You could leave the icons in thier original colour, so that they are easiliy recognizable, but put them on a background coloured like the VM, somewhat like framing the VM's windows in that colour. So for instance, a red and a green firefox would look like that:
![RedFox](https://user-images.githubusercontent.com/22047384/118115268-048c7d80-b3e9-11eb-9f81-acabb3929e8f.jpg)
![GreenFox](https://user-images.githubusercontent.com/22047384/118115300-0f471280-b3e9-11eb-8fc6-c580403c3a37.jpg)

**Comment by ninavizz** [comment]
@GWeck I saw you took the appmenu survey (thank you for that!)... and I did the background-coloration on hover for the _Bessie_ prototype. Yep, background coloring is my preference for how to move forward with the coloration convention—but (understandably) many kinda freaked-out at the idea of not having colorized icons anymore. Baby steps! :)

If the "Favorites Bar" on the left of the menu is ever implemented, I suspect keeping the backgrounds colored not just on hover, but when an app is not being hovered on, will be sought. TBD tho, as that feature is not a priority for a while—and by the time it might be, users will have already adjusted a lot to the new app menu, and other ideas (such as widgets for individual running VMs in the Task Bar) may be in the mix by then. 😃

**Comment by ninavizz** [comment]
@DemiMarie Pretty much everything _you tech types_ do, seems like magic to us _humanbrain/design-y types_... so I bow in reverence to your skills-fu, as well. :)

**Comment by ffsec** [comment]
> ![RedFox](https://user-images.githubusercontent.com/22047384/118115268-048c7d80-b3e9-11eb-9f81-acabb3929e8f.jpg)
> ![GreenFox](https://user-images.githubusercontent.com/22047384/118115300-0f471280-b3e9-11eb-8fc6-c580403c3a37.jpg)

Some time ago the reaction to that view would have been "eye cancer". :wink:

> Yep, background coloring is my preference for how to move forward with the coloration convention—but (understandably) many kinda freaked-out at the idea of not having colorized icons anymore. Baby steps! :)

At first glance this might look like a possible solution, but it isn't.
Icons come in different colors and shapes, so while some might be fine, others won't and the smaller the icon, the worse it gets. At the same time it might open a new attack vector, since the icons are coming straight from the vms. A compromized vm might try to spoof label colors.

For example let's take a completely red icon (32x32), how do you color it if the label color is blue?
If you paint only the transparent background, the blue color won't show at all. Writing complex code, trying to detect shapes and define some area as "paintable background", good luck. So the only possibility would be to shrink the original icon and reserve space around it for the label color. In this example it would result in a large red square with a small blue border around it.
So how much space do you reserve for the border?
1px?
![grafik](https://user-images.githubusercontent.com/31329006/120086646-18fe9480-c0d0-11eb-8fb2-ba8c1f161502.png)
Can you see it if the icon sits on a dark background? Can you tell the color is blue? Or could it also be purple or any other similar color?
Obviously a thicker frame would be needed, which means, you will have to shrink the original icon even further. This will not only result in ugly icons, but is also a huge problem when it comes to smaller icons. How does a 16x16 px icon look like if you shrink it down to add a thick colored border around it? Will you even be able to recognize the icon afterwards? Highly questionable.

Now back to the security risk.
For simplicity I'll just take the icon with the red background from above as example. Let's say a vm with an orange label generates a firefox icon with a red background. If you add the orange border you will get:
1px:
![grafik](https://user-images.githubusercontent.com/31329006/120087075-6da40e80-c0d4-11eb-94fb-a9fc9eb60670.png)
2px:
![grafik](https://user-images.githubusercontent.com/31329006/120087080-801e4800-c0d4-11eb-8287-97f2acd520ae.png)
...

I guess you see where this is going. Just don't do it.

**Comment by ninavizz** [comment]
@ffsec I'm not sure exactly what you're responding to—but, a few quick points. 
1. This issue is for the Applications Tab in the VM Settings window. The topic you're speaking to, is how app icons are colorized by the system. If icon colorization, specifically, is something you'd like to speak to, please create an issue for that—per the Issue Guidelines. Yes, I am a frequent offender of the off-topic segues, but we're all working on getting better about that to keep Issues more focused. :)
2. What @GWeck and I were discussing, was the idea of colorizing the selection bar in the AppMenu, behind an app name and it's corresponding icon. As demonstrated in [the "Bessie" video](https://vimeo.com/542041258) I shared on the AppMenu Survey. So, no—not putting a colored square background shape behind icons. Something else, entirely.
3. Users should also not be making security decisions based on color, alone. If that were a practice we were encouraging, than nothing should be colored by the system. As with the topic of colorizing app icons, that would also need to be a separate Issue.

Thx!

**Comment by ffsec** [comment]
@ninavizz
You wrote "background coloring is my preference for how to move forward with the coloration convention", which suggested a general coloration practice, rather than just background coloration in one particular area. Afterall the OS should have a uniform look and feel instead of having different practices everywhere, so if you colorize the background of the app menu, you should be willing to use this practice for other things aswell.

From reading your comments throughout github, I feel like you can't handle critics very well, is that possible?
I've already evaluated many things you are coming up with now and I know the problems you will run into in many cases. I'm just looking at the larger picture and try to prevent problems before they even come up. I could explain now why background coloration is a bad idea for the app menu background in particular, but I don't really think you'd like to hear it.
So I'll just summarize it with: Don't do it, thanks!

**Comment by ninavizz** [comment]
@ffsec I really do want to have these discussions with you, but I also don't see you reading what I wrote—especially just above, and on another issue. Really, the entire team is making an effort to step-up keeping on topic in Issues, and **I already asked you once in a different issue** to please take the discussion about colorization to #2523 or another related issue. This issue is for the Applications Tab  in the VM Settings window. I realize I am a big historic offender of this, but I am also stepping-up my own game to be better at this.

Defending design decisions is part of the iteration process; it's very different from "getting defensive." If I put up a defense on a decision based on systemic considerations I've looked at (and were not communicated explicitly in an issue) and those defenses still don't hold up, that helps push a _systemic_ decision forward as needed—systemic decisions, often being implied in specific solution decisions. On GitHub people are often responding to _only part_ of a design decision and not the full thing, because of limitations with asynchronous communication and time. That also requires the designer responding to better contextualize a decision, in a defense of it. If that defense still doesn't uphold, that is a necessary step in the critique process. 

I also (personally) do not understand the technical underpinnings of many things, and it can take other people repeating things in different ways for me to understand those things, some of the time. Async collaboration between technical and ux people is far from ideal, and requires patience all around. **Questioning my integrity or anybody else's when your suggestions are not immediately embraced, is just not cool.** 

> I've already evaluated many things you are coming up with now and I know the problems you will run into in many cases. I'm just looking at the larger picture and try to prevent problems before they even come up. 

I'm proposing the background color change _only on hover_. Only one item may be hovered over at a time, and users do not make decisions of a group of items all being hovered upon. No decision will be perfect, and the imperfections of this decision I am ok with; and consider to be far more desirable, than the imperfections of other options. 

Critique is critical to making design decisions better. That is what I seek to foster, here. From what you've written, I do not gather that you've even looked at the videos I shared, to see the fuller proposals.  We all need to do better and to encourage each other to do better. That can be done, but requires patience—and not capping on the integrity of others you don't feel are listening to you.

**Comment by ninavizz** [comment]
**NOTE: Please do not comment any further on this post, unless your comment is about the Applications Tab on the VM Settings Pane.** 

[Contributing Guidelines](https://github.com/QubesOS/qubes-issues/blob/0ee2cc378e576cd2ea249c1a148337a99d3431ae/CONTRIBUTING.md) are clear about keeping issues on-topic, and for all Qubes OS contributors to have issues to act upon to keep things actionable, it is essential that I and all other frequent offenders of this propensity, tighten up about it. 

- **Color topics?** _Comment forth, but on the following (or new) issues..._
  - #6652 to make colorized icons optional
  - #6526 to make colorblind friendly window decorations
  - #2523 to discuss colorization in general, as a lengthy segue to extending Qubes' support for colorization to signify domain properties.

Broader process issues, or "how to ux in GitHub?" I (and others!) welcome discussion on, in Discourse. I also need to get better about being more disciplined in time I spend on Qubes, so will likely only contribute to forum stuff 1-2x per week... but I do welcome the dialogue, there.

**Comment by rysiekpl** [comment]
I would like to say that this is an *extremely* welcome development, especially the easy way to configure launching anything in a DispVM. For a log time I felt this is a missing piece that would make QubesOS much more usable in a safe way.

For example, @freedomofpress [is setting `open-in-dvm.desktop` as the default handler for all mimetypes](https://github.com/freedomofpress/securedrop-debian-packaging/blob/main/securedrop-workstation-config/mimeapps.list.sd-app) for their Debian-based templates. This is a good way of doing this for a locked-down environment like the SecureDrop Workstation, but it's not really user-friendly enough for most regular QubesOS users (and yes, "regular QubesOS user" is a curious term...).

Having such a fine-grained control over things directly in the Qube settings dialogue would be a way better way of dealing with this!

One question: I assume that we're past testing freeze for R4.1-rc3, this is potentially not getting into R4.1? Or could it be expected to show up there anyway even if developed later?

Many thanks for this wonderful work!

**Comment by marmarek** [comment]
> One question: I assume that we're past testing freeze for R4.1-rc3, this is potentially not getting into R4.1? Or could it be expected to show up there anyway even if developed later?

you're right, it's too late to get this into R4.1. But the backend of this features made it, it's just missing documentation (https://github.com/QubesOS/qubes-issues/issues/7130).

**Comment by rysiekpl** [comment]
> you're right, it's too late to get this into R4.1.

Dang. Thanks for making it clear though.

> But the backend of this features made it, it's just missing documentation (#7130).

Right. Salt to the rescue, I guess!

**Comment by TNTBOMBOM** [comment]
If possible, an additional feature that could be beneficial to the application tab is a search functionality. Since multiple applications already exist within one VM, users could simply input the first letter of the desired application in the search bar, which would then narrow down the list of available applications for the user.

**Comment by alimirjamali** [comment]
> something really useful would be to have an autostart checkbox

We discussed this with @kennethrrosen two weeks ago on forum and shared some design ideas about it:

https://forum.qubes-os.org/t/autostart-applications-in-appvm-settings-feature-request/30703/5

Technically in he proposed design sketches for new Appmenu desing, theoretically it would be possible to add another column for autostart apps.

**Comment by alimirjamali** [comment]
> If possible, an additional feature that could be beneficial to the application tab is a search functionality. Since multiple applications already exist within one VM, users could simply input the first letter of the desired application in the search bar, which would then narrow down the list of available applications for the user.

The above will be implemented in https://github.com/QubesOS/qubes-issues/issues/10276

**Comment by ben-grande** [comment]
The opening in disposable button would be related https://github.com/qubesos/qubes-issues/issues/441
